osutil/sync_dir.go

```diff
@@ -51,10 +52,16 @@ var errSameState = fmt.Errorf("file state has not changed")
 // the directory.  Map keys must be file names relative to the directory.
 // Sub-directories in the name are not allowed.
 //
-// The function stops at the first encountered error but reports all of the
-// changes performed so far. Information about the performed changes is
-// returned to the caller for any extra processing that might be required (e.g.
-// to run some helper program).
+// If writing any of the files fails, EnsureDirState switches to erase mode
+// where *all* of the files managed by the glob pattern are removed (including
+// those that may have been already written). The return value is an empty list
+// of changed files, the real list of removed files and the first error.
```

Comment:
`..., the list of removed files including those that were previously written and then removed, and the first error.`

This behavior also needs to be properly used. Right now profiles are not being unloaded if there are errors.